Source/WebCore/platform/graphics/PlatformDisplay.cpp

```diff
@@ -244,67 +229,36 @@ EGLDisplay PlatformDisplay::eglDisplay() const
 {
     if (!m_eglDisplayInitialized)
         const_cast<PlatformDisplay*>(this)->initializeEGLDisplay();
-    return m_eglDisplay;
+    return m_eglDisplay ? m_eglDisplay->eglDisplay() : EGL_NO_DISPLAY;
 }
 
 bool PlatformDisplay::eglCheckVersion(int major, int minor) const
 {
     if (!m_eglDisplayInitialized)
         const_cast<PlatformDisplay*>(this)->initializeEGLDisplay();
-
-    return (m_eglMajorVersion > major) || ((m_eglMajorVersion == major) && (m_eglMinorVersion >= minor));
+    return m_eglDisplay ? m_eglDisplay->checkVersion(major, minor) : false;
 }
 
-const PlatformDisplay::EGLExtensions& PlatformDisplay::eglExtensions() const
+const GLDisplay::Extensions& PlatformDisplay::eglExtensions() const
 {
     if (!m_eglDisplayInitialized)
         const_cast<PlatformDisplay*>(this)->initializeEGLDisplay();
-    return m_eglExtensions;
+    static GLDisplay::Extensions empty;
```

Comment:
This can be `static const`:

```suggestion
    static const GLDisplay::Extensions empty;
```

Reply:
This will be removed in the next patch.